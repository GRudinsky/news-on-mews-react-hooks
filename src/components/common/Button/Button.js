import React from 'react'

export default function Button(props) {
  return (
    <div>
<button data-testid="button" className="categoryButton" onClick={props.onClick} value={props.itle}>{props.title}</button>
    </div>
  )
}
