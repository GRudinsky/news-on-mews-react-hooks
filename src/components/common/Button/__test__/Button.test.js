import React from 'react'
import ReactDOM from 'react-dom'
import Button from './../Button'
import {render, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'

afterEach(cleanup)
it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<Button />, div)
})

it("renders button correctly", () => {
  const { getByTestId } = render(<Button title={"Technology"} />)
  expect(getByTestId("button")).toHaveTextContent("Technology")
  })

it("matches snapshot", () => {
  const tree = renderer.create(<Button title={"Technology"}/>).toJSON()
  expect(tree).toMatchSnapshot()
})

  
  
  
  
  
