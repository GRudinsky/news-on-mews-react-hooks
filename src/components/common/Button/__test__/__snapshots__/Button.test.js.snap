// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <button
    className="categoryButton"
    data-testid="button"
  >
    Technology
  </button>
</div>
`;
